lua/ttt2/libraries/bodysearch.lua

```diff
@@ -930,11 +945,23 @@ if CLIENT then
             -- only use the first text entry here
             local transText = LANG.GetDynamicTranslation(text[1].body, text[1].params, true)
 
-            search[type] = {
-                img = searchData.iconMaterial:GetName(),
-                text = transText,
-                p = i, -- sorting number
-            }
+            if searchData.iconMaterial then
+                -- note: GetName only returns the material name. This fails if the addon uses a
+                -- png for its material, we therefore have to check if the material exists on disk
+                local materialFile = searchData.iconMaterial:GetName()
+
+                if file.Exists("materials/" .. materialFile .. ".png", "GAME") then
+                    materialFile = materialFile .. ".png"
+                end
+
+                search[type] = {
+                    img = materialFile,
+                    text = transText,
+                    p = index, -- sorting number
+                }
+
+                index = index + 1
+            end
```

Comment:
2 things here:

1. You are now not showing anything when no icon is available, maybe we should just display some kind of fallback? or at least display this one without an icon?
2. This special png handling should not be needed. Because DImage, which is used in SetIcon in SimpleIcon does already turn the string into a Material, which should work automatically with .png. See https://github.com/Facepunch/garrysmod/blob/master/garrysmod/lua/vgui/dimage.lua#L91

Reply:
1. But what should I show if the weapon has no icon and also no inflictor? The missing icon material?
2. But it did not work when I did pass it without an .png, so who knows

Reply:
Done the first thing, not sure what I could do about the second

Reply:
Why is the searchData even providing the material instead of the filename? Seems like we may be able to change that?

Reply:
because we cache the materials and I don't want to recreate the materials based on their file

Reply:
I'd rather have this fixed correctly instead of only for png images, so I'd suggest adjusting the scoreboard code.
But i would also check this out on my own and see if i can find a solution, if you don't want to.

Reply:
I see, can you then just add cases for jpeg and jpg, i think these are currently the only 2 other types that are allowed in the addon whitelist.

Reply:
As a disclaimer: I don't know if any addon is actually using this information, but it could be

Reply:
I considered jpeg but couldn't think of a single case I've seen jpegs used for icons. But I can add this

Reply:
Also note, this whole function here is removed as soon as we have a new scoreboard. This only converts the new bodysearch data to the legacy table

Reply:
You've missed ".jpeg"